### PR TITLE
coin and x/cash: tests cleanup 

### PR DESCRIPTION
**Pull request by husio**
coin and x/cash: tests cleanup  

part of issue #249

### CODE REVIEW

File: coin/coin_test.go [hunk 1]
```diff
@@ -7,148 +7,158 @@ import (
...
 
-	for idx, tc := range cases {
-		t.Run(fmt.Sprintf("case-%d", idx), func(t *testing.T) {
-			// make sure both show proper results
-			assert.Equal(t, tc.a.IsZero(), tc.aState == zero)
-			assert.Equal(t, tc.a.IsPositive(), tc.aState == pos)
-			assert.Equal(t, !tc.a.IsNonNegative(), tc.aState == neg)
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			res := tc.a.Compare(tc.b)
```
Comment (ethanfrey) [inline coin/coin_test.go]: Why remove the IsZero/IsPositive/IsNonNegative tests?
You could move them into another test case (this one is doing two checks at once),
but it is good to ensure those cases are covered (maybe nicer than here)

File: coin/coin_test.go [hunk 2]
```diff
@@ -7,148 +7,158 @@ import (
...
 
-			// make sure compare is correct
-			assert.Equal(t, tc.a.Compare(tc.b), tc.expect)
+	assert.Equal(t, a.Ticker, n.Ticker)
+	assert.Equal(t, a.Whole, -n.Whole)
+	assert.Equal(t, a.Fractional, -n.Fractional)
 
-			assert.True(t, tc.a.SameType(tc.b))
-		})
+	if nn := a.Negative().Negative(); !a.Equals(nn) {
```
Comment (ethanfrey) [inline coin/coin_test.go]: Good test

File: coin/coin_test.go (hunk 1 above)
Comment (ethanfrey) [reply coin/coin_test.go]: I only see `TestCoinIsPositive` below

File: coin/coin_test.go [hunk 3]
```diff
@@ -7,148 +7,158 @@ import (
...
+		wantNormValErr       *errors.Error
+	}{
+		"valid coin with a negative fractional": {
+			coin:                 NewCoin(0, -100, "DIN"),
+			wantValErr:           nil,
+			wantNormalized:       NewCoin(0, -100, "DIN"),
+			wantNormalizationErr: nil,
+			wantNormValErr:       nil,
+		},
+		"interger and fraction with different sign": {
```
Comment (ethanfrey) [inline coin/coin_test.go]: interger -> integer

File: x/cash/controller_test.go [hunk 4]
```diff
@@ -1,45 +1,43 @@
...
 
-func getWallet(kv weave.KVStore, addr weave.Address) coin.Coins {
+func wallet(t testing.TB, kv weave.KVStore, addr weave.Address) coin.Coins {
+	t.Helper()
+
 	bucket := NewBucket()
 	res, err := bucket.Get(kv, addr)
 	if err != nil {
-		panic(err) // testing only
+		t.Fatalf("cannot get wallet for %q: %s", addr, err)
```
Comment (ethanfrey) [inline x/cash/controller_test.go]: better.
even better would be `assert.Nil(t, err)`

File: x/cash/controller_test.go [hunk 5]
```diff
@@ -1,45 +1,43 @@
...
 	return AsCoins(res)
 }
 
 type issueCmd struct {
-	addr   weave.Address
-	amount coin.Coin
-	isErr  bool
+	addr    weave.Address
+	amount  coin.Coin
+	wantErr *errors.Error
```
Comment (ethanfrey) [inline x/cash/controller_test.go]: :+1:

File: coin/coin_test.go (hunk 1 above)
Comment (husio) [reply coin/coin_test.go]: I will add separate test functions to test them.

File: x/cash/controller_test.go (hunk 4 above)
Comment (husio) [reply x/cash/controller_test.go]: `assert.Nil` will not print the address, which might be helpful during the debugging.

**Pull request merged**